// Copyright 2021 Google LLC

// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at

//     https://www.apache.org/licenses/LICENSE-2.0

// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package pyproto.test;

import "google/protobuf/any.proto";

message Topping {
  string name = 1;
}

message MatchaMilkTea {
  string name = 1;
  float price = 2;
  string seller = 3;
  google.protobuf.Any topping1 = 4;
  google.protobuf.Any topping2 = 5;
  Topping topping3 = 6;
}

message GreenTeaMilkTea {
  string name = 1;
  int64 price = 2;
  string seller = 3;
  Topping topping1 = 4;
  google.protobuf.Any topping2 = 5;
  google.protobuf.Any topping3 = 6;
}
